nixos/modules/services/games/archisteamfarm.nix

```
@@ -161,31 +161,33 @@ in
   };
 
   config = mkIf cfg.enable {
+    # TODO: drop with 24.11
+    services.archisteamfarm.dataDir = lib.mkIf (lib.versionAtLeast config.system.stateVersion "24.05") "/var/lib/asf";
```

```suggestion
    services.archisteamfarm.dataDir = mkIf (lib.versionAtLeast config.system.stateVersion "24.05") (mkDefault "/var/lib/asf");
```
Use `mkDefault` so that preexisting configs configs that override dataDir continue to function.
Also removed the `lib.` as the file already contains a `with lib;` at the top.

---

Sure. Maybe I should drop that option entirely in another PR since I am not regularly testing it.

Also dropped with lib, since it is discouraged anyway.